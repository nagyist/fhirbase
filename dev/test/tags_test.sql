--db:fhirb
SET escape_string_warning=off;
--{{{


\set old_tags '[{"scheme": "http://pt.com", "term": "http://pt/vip", "label":"pt"}]'
\set new_tags '[{"scheme": "http://pt2.com", "term": "http://pt/vip2", "label":"pt2"}]'
\set more_tags '[{"scheme": "http://pt3.com", "term": "http://pt/vip3", "label":"pt3"}]'


BEGIN;

SELECT 'insert' FROM fhir_create('{}'::jsonb, 'Patient', '{"resourceType":"Patient"}'::jsonb, :'old_tags'::jsonb);

SELECT assert_eq(
  :'old_tags'::jsonb,
  (
    SELECT x->'category'
    FROM fhir_tags('{}'::jsonb, 'Patient', (SELECT logical_id from patient)) x
),'fhir_tags');

SELECT fhir_affix_tags('{}'::jsonb,
  'Patient',
  (SELECT logical_id from patient),
  :'new_tags'::jsonb
);

SELECT assert_eq(
  _merge_tags(:'new_tags'::jsonb, :'old_tags'),
  (
    SELECT x->'category'
    FROM fhir_tags('{}'::jsonb, 'Patient', (SELECT logical_id from patient)) x
),'fhir_tags');

SELECT 'update' FROM (
  SELECT fhir_update('{}'::jsonb,
    'Patient'::text,
    logical_id::text,
    version_id::text,
    '{"resourceType":"Patient"}'::jsonb,
    :'new_tags'::jsonb)
  FROM patient
) _ ;


SELECT fhir_tags('{}'::jsonb, 'Patient'::text, logical_id, version_id)
FROM patient_history
LIMIT 1;


SELECT fhir_affix_tags('{}'::jsonb, 'Patient'::text, logical_id, version_id, :'more_tags'::jsonb)
FROM patient_history
LIMIT 1;

SELECT fhir_tags('{}'::jsonb, 'Patient'::text, logical_id, version_id)
FROM patient_history
LIMIT 1;

SELECT fhir_remove_tags('{}'::jsonb, 'Patient'::text, logical_id)
FROM patient
LIMIT 1;

SELECT fhir_tags('{}'::jsonb, 'Patient'::text, logical_id)
FROM patient
LIMIT 1;

SELECT fhir_remove_tags('{}'::jsonb, 'Patient'::text, logical_id, version_id)
FROM patient_history
LIMIT 1;

SELECT fhir_tags('{}'::jsonb, 'Patient'::text, logical_id, version_id)
FROM patient_history
LIMIT 1;

ROLLBACK;

--}}}
