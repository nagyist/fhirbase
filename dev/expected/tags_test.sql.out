SET escape_string_warning=off;
SET
BEGIN;
BEGIN
SELECT 'insert' FROM fhir_create('{}'::jsonb, 'Patient', '{"resourceType":"Patient"}'::jsonb, '[{"scheme": "http://pt.com", "term": "http://pt/vip", "label":"pt"}]'::jsonb);
 ?column? 
----------
 insert
(1 row)

SELECT assert_eq(
  '[{"scheme": "http://pt.com", "term": "http://pt/vip", "label":"pt"}]'::jsonb,
  (
    SELECT x->'category'
    FROM fhir_tags('{}'::jsonb, 'Patient', (SELECT logical_id from patient)) x
),'fhir_tags');
  assert_eq   
--------------
 OK fhir_tags
(1 row)

SELECT fhir_affix_tags('{}'::jsonb,
  'Patient',
  (SELECT logical_id from patient),
  '[{"scheme": "http://pt2.com", "term": "http://pt/vip2", "label":"pt2"}]'::jsonb
);
                                                                fhir_affix_tags                                                                
-----------------------------------------------------------------------------------------------------------------------------------------------
 [{"term": "http://pt/vip", "label": "pt", "scheme": "http://pt.com"}, {"term": "http://pt/vip2", "label": "pt2", "scheme": "http://pt2.com"}]
(1 row)

SELECT assert_eq(
  _merge_tags('[{"scheme": "http://pt2.com", "term": "http://pt/vip2", "label":"pt2"}]'::jsonb, '[{"scheme": "http://pt.com", "term": "http://pt/vip", "label":"pt"}]'),
  (
    SELECT x->'category'
    FROM fhir_tags('{}'::jsonb, 'Patient', (SELECT logical_id from patient)) x
),'fhir_tags');
  assert_eq   
--------------
 OK fhir_tags
(1 row)

SELECT 'update' FROM (
  SELECT fhir_update('{}'::jsonb,
    'Patient'::text,
    logical_id::text,
    version_id::text,
    '{"resourceType":"Patient"}'::jsonb,
    '[{"scheme": "http://pt2.com", "term": "http://pt/vip2", "label":"pt2"}]'::jsonb)
  FROM patient
) _ ;
 ?column? 
----------
 update
(1 row)

SELECT fhir_tags('{}'::jsonb, 'Patient'::text, logical_id, version_id)
FROM patient_history
LIMIT 1;
                                                                                       fhir_tags                                                                                        
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {"category": [{"term": "http://pt/vip", "label": "pt", "scheme": "http://pt.com"}, {"term": "http://pt/vip2", "label": "pt2", "scheme": "http://pt2.com"}], "resourceType": "TagList"}
(1 row)

SELECT fhir_affix_tags('{}'::jsonb, 'Patient'::text, logical_id, version_id, '[{"scheme": "http://pt3.com", "term": "http://pt/vip3", "label":"pt3"}]'::jsonb)
FROM patient_history
LIMIT 1;
                                                                                                    fhir_affix_tags                                                                                                    
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 [{"term": "http://pt/vip", "label": "pt", "scheme": "http://pt.com"}, {"term": "http://pt/vip2", "label": "pt2", "scheme": "http://pt2.com"}, {"term": "http://pt/vip3", "label": "pt3", "scheme": "http://pt3.com"}]
(1 row)

SELECT fhir_tags('{}'::jsonb, 'Patient'::text, logical_id, version_id)
FROM patient_history
LIMIT 1;
                                                                                                                           fhir_tags                                                                                                                            
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {"category": [{"term": "http://pt/vip", "label": "pt", "scheme": "http://pt.com"}, {"term": "http://pt/vip2", "label": "pt2", "scheme": "http://pt2.com"}, {"term": "http://pt/vip3", "label": "pt3", "scheme": "http://pt3.com"}], "resourceType": "TagList"}
(1 row)

SELECT fhir_remove_tags('{}'::jsonb, 'Patient'::text, logical_id)
FROM patient
LIMIT 1;
 fhir_remove_tags 
------------------
                1
(1 row)

SELECT fhir_tags('{}'::jsonb, 'Patient'::text, logical_id)
FROM patient
LIMIT 1;
                  fhir_tags                  
---------------------------------------------
 {"category": [], "resourceType": "TagList"}
(1 row)

SELECT fhir_remove_tags('{}'::jsonb, 'Patient'::text, logical_id, version_id)
FROM patient_history
LIMIT 1;
 fhir_remove_tags 
------------------
                1
(1 row)

SELECT fhir_tags('{}'::jsonb, 'Patient'::text, logical_id, version_id)
FROM patient_history
LIMIT 1;
                  fhir_tags                  
---------------------------------------------
 {"category": [], "resourceType": "TagList"}
(1 row)

ROLLBACK;
ROLLBACK
